numpy/_core/src/multiarray/stringdtype/dtype.h

```diff
@@ -49,6 +49,9 @@ stringdtype_finalize_descr(PyArray_Descr *dtype);
 NPY_NO_EXPORT int
 _eq_comparison(int scoerce, int ocoerce, PyObject *sna, PyObject *ona);
 
+NPY_NO_EXPORT PyArray_StringDTypeObject *
+common_instance(PyArray_StringDTypeObject *dtype1, PyArray_StringDTypeObject *dtype2);
```

Comment:
might make sense to name it `string_common_instance` or so if we export it, in case this might end up getting included elsewhere?  (just a passing thought)